porting/building-ubports-boot.rst

```diff
@@ -0,0 +1,95 @@
+
+Building ubports-boot
+=====================
+
+Ubuntu Touch uses Upstart rather than SystemD for its init daemon. Because of this, it is not fully Halium-compatible and is not able to use the vanilla hybris-boot that Halium produces. For this reason, we need to build ubports-boot.
+
+
+Add source to Halium tree
+-------------------------
+
+The first thing that needs to be done is adding the ubports-boot source to your Halium tree. You may choose to do this by adding it to your local manifests (recommended) or simply cloning it in place.
+
+.. warning::
+
+    Due to a `bug in the ubports-boot build scripts <https://github.com/ubports/ubports-boot/issues/3>`_, you will need to remove it from your tree before you're able to build ``hybris-boot`` again.
+
+
+Add to local manifest
+^^^^^^^^^^^^^^^^^^^^^
+
+Create a new file in ``.repo/local_manifests`` called ``ubports-boot.xml``. Add the following to the file:
+
+.. code-block:: xml
+
+    <?xml version="1.0" encoding="UTF-8"?>
+    <manifest>
+        <project name="ubports/ubports-boot" path="halium/ubports-boot" remote="hal" revision="master" />
+    </manifest>
+
+Now, just do a ``repo sync`` to download the new source.
+
+
+Manual clone
+^^^^^^^^^^^^
+
+You may also choose to clone `the ubports-boot repository <https://github.com/ubports/ubports-boot>`_ into your tree manually. It should be placed into ``BUILDDIR/halium/ubports-boot``.
+
+If you share your code and build instructions, please note that you've done this.
+
+
+Fix mounts
+----------
+
+.. todo::
+
+    This is a workaround for `ubports/ubports-boot #1 <https://github.com/ubports/ubports-boot/issues/1>`_ and `ubports/ubports-boot #2 <https://github.com/ubports/ubports-boot/issues/2>`_. It should not be considered a permanent fix. Be prepared to revert this later when these bugs have been fixed.
+
+The initramfs of the ubports-boot image takes a commandline parameter, ``datapart=``, which points it to the block device that is normally mounted at ``/data`` in Android. This parameter is automatically embedded into the image at build time by finding ``/data`` in your device's default fstab and using its source. Unfortunately, this source is generally a node in ``/dev/block/bootdevice/by-name`` or something else strange, which is only available in an Android initramfs. To fix this, Halium uses the fixup-mountpoints script, which you are probably familiar with by now.
+
+The ubports-boot initramfs does not use the fixup-mountpoints script, nor does it put block devices in the proper place for mounting without modification. For this reason, we are going to have to edit your device's fstab.
+
+The first step to this process is figuring out where your fstab actually is. For most, this is inside ``$BUILDDIR/device/[manufacturer]/[codename]/rootdir/etc`` and it is named either ``fstab.qcom`` or ``fstab.devicename``. Open the file for editing.
```

Comment:
Let's not use too many different styles _variables_ in the instructions. I'd suggest ALLCAPS, but any consistent style is fine. Here you use three different ones:

- with a dollar sign $BUILDDIR
- with rectangular brackets [manufacturer]
- without any emphasis in fstab.devicename

Also, I think the appendix to fstab is the same as the foldername, so in conclusion I suggest: 

this is inside ``BUILDDIR/device/MANUFACTURER/CODENAME/rootdir/etc`` and it is named either ``fstab.qcom`` or ``fstab.CODENAME``.